### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by semperos**
I know these aren't the best or most exhaustive tests, but if this is acceptable, I have a small follow-on patch for #4 to push as well. Didn't want to make a change without tests in place to mitigate breaking things.

Unit tests:

```
go test
```

Unit + system tests:

```
go test -system
```

### CODE REVIEW

File: rpenv_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			arg: "qa",
+			uri: "http://ag-web-01.qa.atl.primedia.com/ops/env/environment.cfg",
+		},
+		{
+			arg: "prod",
+			uri: "http://ag-web-01.atl.primedia.com/ops/env/environment.cfg",
+		},
+		{
+			arg: "production",
+			uri: "http://ag-web-01.atl.primedia.com/ops/env/environment.cfg",
```
Comment (colinrymer) [inline rpenv_test.go]: I'm not thrilled about duplicating the URLs between here and the implementation file.

Thoughts on making the URLs constants and checking that we return the appropriate constant?

File: rpenv_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		t.Errorf("Command 'ls' did not return a 0 exit code. Are you on a *nix machine?")
+	}
+	cmd := []string{"ci", "ls", "not-here-1234"}
+	exitCode = executeCommand(cmd, []string{})
+	if exitCode != 1 {
+		t.Errorf("Command '%q' returned %q exit code, but expected %q.", cmd, exitCode, 0)
+	}
+
+	restoreStd(writerStdout, writerStderr, stdout, stderr)
+}
```
Comment (colinrymer) [inline rpenv_test.go]: Why should `executeComand` not return a non-zero exit code?

File: rpenv_test.go (hunk 1 above)
Comment (semperos) [inline rpenv_test.go]: That'd be fine by me. Started with simple "contains" checks for the subdomains but that doesn't cover the `"prod"` code branch. I'll whip something up.

File: rpenv_test.go (hunk 2 above)
Comment (semperos) [inline rpenv_test.go]: Two cases in this one, one for `ls` which should return `0`, and one for `ls not-here-1234` which should return `1`. It demonstrates that `executeCommand` in fact returns the actual response code and doesn't end up always returning `0` (the happy path).

**Comment by rawsyntax** [comment]
@semperos if you add your patch addressing #4 here I can test it by actually using it

**Comment by semperos** [comment]
@rawsyntax Will do, I'll add the commit to this branch. If folks don't want the two coupled in a single PR, I can always rebase and take it out again.

**Comment by semperos** [comment]
Improvement (related to #4) pushed.

**Comment by rawsyntax** [comment]
I still get the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x388f]

goroutine 1 [running]:
panic(0x330f40, 0xc820014110)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
main.httpRequestBodyAsString(0x417e00, 0x3c, 0x0, 0x0)
	/Users/ehimmelreich/source/rpenv/rpenv.go:106 +0x45f
main.envVars(0x417e00, 0x3c, 0x0, 0x0, 0x0)
	/Users/ehimmelreich/source/rpenv/rpenv.go:56 +0x5a
main.main()
	/Users/ehimmelreich/source/rpenv/rpenv.go:26 +0x114
```

when not on a vpn for those urls

**Comment by semperos** [comment]
@rawsyntax Good catch, I'm on it. Thanks for checking, I hadn't tried off the VPN.

**Comment by semperos** [comment]
@rawsyntax Here's what I see when I run it with my VPN turned off:

```
go run rpenv.go ci
# Delay as it tries to resolve...
Unable to reach URL http://ag-web-01.ci.atl.primedia.com/ops/env/environment.cfg
Please contact infra to remedy this
exit status 3
```

Thoughts?

**Comment by semperos** [comment]
And here's test output with VPN turned off:

```
λ: go test -v -system
=== RUN   TestEnvUri
--- PASS: TestEnvUri (0.00s)
=== RUN   TestEnvVars
Unable to reach URL http://ag-web-01.ci.atl.primedia.com/ops/env/environment.cfg
Please contact infra to remedy this
exit status 3
FAIL	github.com/rentpath/rpenv	0.012s
```

**Comment by rawsyntax** [comment]
working for me now 🚢

**Pull request merged**